software/examples/snap_example_set.c

```diff
@@ -292,6 +292,15 @@ int main(int argc, char *argv[])
 			break;
 		case 's':	/* size */
 			size = strtol(optarg, (char **)NULL, 0);
+			if ((errno == ERANGE && (size == LONG_MAX || size == LONG_MIN))
+                   || (errno != 0 && size == 0)) {
```

Comment:
the || goes into the line above. Mind the 80 columns. Maybe use the function where you can do the MiB, KiB, ... etc.